anger.js

```diff
@@ -36,7 +36,7 @@ function anger (opts) {
   }
 
   steed.each(clients, (client, done) => {
-    client.connect({ auth: auth(client, client['_angerIndex']) }, done)
+    client.connect({ auth: typeof auth === 'function' ? auth(client, client['_angerIndex']) : auth }, done)
```

Comment:
I think you can wrap the original `auth` parameter in a function, it would be more functional and nicer to read.

Also, you can use `client._angerIndex`.